CodecheckPlugin.php

```diff
@@ -1,178 +1,221 @@
+                // Check if CODECHECK options are already in the form
+                foreach ($form->fields as $field) {
+                    if ($field instanceof \PKP\components\forms\FieldOptions && $field->name === 'genreId') {
+                        // Check if CODECHECK options already exist
+                        $hasCodecheckOptions = false;
+                        foreach ($field->options as $option) {
+                            if (str_contains($option['label'], 'CODECHECK') || str_contains($option['label'], 'codecheck')) {
+                                $hasCodecheckOptions = true;
+                                break;
+                            }
                         }
-                    }
-                    
-                    if ($hasCodecheckOptions) {
-                        error_log("CODECHECK: Options already exist in form, skipping");
-                        return false;
-                    }
-                    
-                    // Create genres if needed (only once)
-                    $this->createCodecheckGenres();
-                    
-                    // Get fresh genres from database
-                    $context = $request->getContext();
-                    $genreDao = \PKP\db\DAORegistry::getDAO('GenreDAO');
-                    $genres = $genreDao->getByContextId($context->getId());
-                    
-                    $codecheckGenres = [];
-                    while ($genre = $genres->next()) {
-                        $name = $genre->getLocalizedName();
-                        if (is_array($name)) {
-                            $name = $name['en'] ?? $name[array_key_first($name)] ?? '';
+                        
+                        if ($hasCodecheckOptions) {
+                            return false;
                         }
                         
-                        if (str_contains($name, 'CODECHECK') || str_contains($name, 'codecheck')) {
-                            $codecheckGenres[] = [
-                                'value' => $genre->getId(),
-                                'label' => $name
-                            ];
+                        $this->createCodecheckGenres();
+                        
+                        $context = $request->getContext();
+                        $genreDao = \PKP\db\DAORegistry::getDAO('GenreDAO');
+                        $genres = $genreDao->getByContextId($context->getId());
+                        
+                        $codecheckGenres = [];
+                        while ($genre = $genres->next()) {
+                            $name = $genre->getLocalizedName();
+                            if (is_array($name)) {
+                                $name = $name['en'] ?? $name[array_key_first($name)] ?? '';
+                            }
+                            
+                            if (str_contains($name, 'CODECHECK') || str_contains($name, 'codecheck')) {
```

Comment:
I think it would be more readable here to transform $name to lowercase and then make a contains check only once, see `strtolower`.